python_ta/cfg/graph.py

```diff
@@ -274,11 +277,32 @@ def update_edge_z3_constraints(self) -> None:
 
                 # traverse into target node
                 for node in edge.target.statements:
-                    if isinstance(node, Assign):
-                        # mark reassigned variables
-                        for target in node.targets:
-                            if isinstance(target, AssignName):
-                                z3_environment.assign(target.name)
+                    if isinstance(node, (Assign, AugAssign)):
+                        self._handle_variable_reassignment(node, z3_environment)
+
+    def _handle_variable_reassignment(self, node: NodeNG, env: Z3Environment) -> None:
+        """Check for reassignment statements and invoke Z3 environment"""
+        if isinstance(node, Assign):
+            for target in node.targets:
+                if isinstance(target, AssignName):
+                    env.assign(target.name)
+        elif isinstance(node, AugAssign):
+            env.assign(node.target.name)
+
+    def update_edge_feasibility(self) -> None:
+        """Traverse through edges in DFS order and update is_feasible
+        attribute of each edge. Edges that are unreachable with the given
+        set of Z3 constraints will have is_feasible set to False
+        """
+        if not z3_dependency_available:
+            return
+
+        for path_id, path in enumerate(self.get_paths()):
+            solver = z3.Solver()
+            for edge in path:
+                solver.add(edge.z3_constraints[path_id])
+                if solver.check() == z3.sat:
+                    edge.is_feasible = True
```

Comment:
Overall, I'd like the default value of `edge.is_feasible` to be `True` (set in the initializer), and then set to `False` when the Z3 solver gives "unsatisfiable" for every path that uses the edge.

This is the more conservative default value, in the sense that later on, we'll modify checks to ignore edges that are infeasible, so we want to avoid at all costs the error of marking an edge as infeasible when it's actually feasible.

Your current approach makes the updating a bit awkward because you're iterating across all paths, and then calling `solver.check` for each path's edge. You should be able to do it the other way, iterating across all *edges* and then calling `solver.check` for each edge's set of path constraints. ("if all constraints are unsatisfiable, `edge.is_feasible = False`")